test/common/parse-util_test.c

```diff
@@ -0,0 +1,38 @@
+#include "../../src/common/memory.h"
+#include "../../src/common/parse-util.h"
+
+#include <assert.h>
+#include <stdio.h>
+
+#ifndef __FUNCTION_NAME__
+#        define __FUNCTION_NAME__ __func__
+#endif
+
+void test_parse_long(char *input, int expectedReturn, long expectedResult) {
```

Comment:
For C safety, we should check function arguments that are pointers for NULL.  In this function, if `input` is NULL, the program will crash with the call to asprintf().

Reply:
Good catch! Added the `NULL` as test case and fixed the bug.